src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php

```diff
@@ -356,216 +446,192 @@ public function getShutdownOnException()
+    public function setEventDispatcher(EventDispatcherInterface $dispatcher = null)
     {
-        foreach ($this->iterationsIntervalCallbacks as $iterationsIntervalCallback) {
-            if (($this->getLoopCount() + 1) % $iterationsIntervalCallback['interval'] === 0) {
-                call_user_func($iterationsIntervalCallback['callable'], $input, $output);
-            }
-        }
-    }
+        $this->dispatcher = $dispatcher;
 
-    protected function setup(InputInterface $input, OutputInterface $output)
-    {
+        return $this;
     }
 
-    protected function tearDown(InputInterface $input, OutputInterface $output)
+    /**
+     * Return the last exception.
+     *
+     * @return \Exception|null
+     */
+    public function getLastException()
     {
+        return $this->lastException;
     }
 
     /**
-     * Return the command name.
+     * Set the last exception.
      *
-     * @return string
+     * @param \Exception $e
+     *
+     * @return \M6Web\Bundle\DaemonBundle\Command\DaemonCommand
      */
-    public function __toString()
+    protected function setLastException(\Exception $e)
     {
-        return str_replace(':', '-', $this->getName());
+        $this->lastException = $e;
+
+        return $this;
     }
 
     /**
-     * Retrieve configured events.
+     * Add your own callback after every iteration interval.
      *
-     * @return DaemonCommand
+     * @param int $iterationsInterval
+     * @param callable $onIterationsInterval
      */
-    protected function configureEvents()
+    public function addIterationsIntervalCallback($iterationsInterval, callable $onIterationsInterval)
     {
-        $container = $this->getContainer();
-        $key = 'm6web_daemon.iterations_events';
-
-        if ($container->hasParameter($key)) {
-            $this->iterationsEvents = $container->getParameter($key);
+        if (!is_int($iterationsInterval) || ($iterationsInterval <= 0)) {
+            throw new \InvalidArgumentException('Iteration interval must be a positive integer');
         }
 
-        return $this;
+        $this->iterationsIntervalCallbacks[] = ['interval' => $iterationsInterval, 'callable' => $onIterationsInterval,];
     }
 
     /**
-     * Dispatch configured events.
+     * Return the command name.
      *
-     * @return DaemonCommand
+     * @return string
      */
-    protected function dispatchConfigurationEvents()
+    public function __toString()
     {
-        foreach ($this->iterationsEvents as $event) {
-            if (!($this->loopCount % $event['count'])) {
-                $this->dispatchEvent($event['name']);
-            }
-        }
-
-        return $this;
+        return str_replace(':', '-', $this->getName());
     }
 
     /**
-     * Add daemon options to the command definition.
+     * Set the sleeping time (used between two loops).
      */
-    private function configureDaemonDefinition()
+    protected function setSleepingTime(int $µseconds): void
```

Comment:
I don't see any code nor configuration using this method

Is it meant to be used as is?

```php
class MyCommand extends DaemonCommand {
  public function __construct() {
    $this->setSleepingTime(1000000); // 1 second
  }
}
```

Reply:
Not in the constructor : the sleeping time is only for the next iteration - maybe we could name this method setNexItirationSleepingTime (if you have a better name !) - the idea is to replace an "usleep in a worker to give the cpu to react loop instead of simply waiting/sleeping between two iteration.

@DocRoms We need to explain in ReadMe and change the name of setSleepingTime.